org.lflang/src/org/lflang/ASTUtils.xtend

```diff
@@ -575,6 +575,19 @@ class ASTUtils {
      * @return Textual representation of the given argument.
      */
     def static String toText(Code code) {
+        return CodeMap.Correspondence.tag(code, toUntaggedText(code), true)
+    }
+
+    /**
+     * Translate the given code into its textual representation
+     * without any {@code CodeMap.Correspondence} tags inserted.
+     * @param code AST node to render as string.
+     * @return Textual representation of the given argument.
+     */
+    def private static String toUntaggedText(Code code) {
+        // FIXME: This function should not be necessary, but it is because we currently
```

Comment:
I need to fix this comment because when I wrote "parse", I should have said "inspect". The problem I alluded to in this FIXME was not really parsing, but rather the use of regexes to process the code blocks.

Reply:
OK, gotcha. Yes, I misunderstood what you meant there :-)